packages/flutter/lib/src/cupertino/switch.dart

```diff
@@ -141,7 +151,9 @@ class CupertinoSwitch extends StatefulWidget {
     this.autofocus = false,
     this.dragStartBehavior = DragStartBehavior.start,
   })  : assert(activeThumbImage != null || onActiveThumbImageError == null),
-        assert(inactiveThumbImage != null || onInactiveThumbImageError == null);
+        assert(inactiveThumbImage != null || onInactiveThumbImageError == null),
+        _activeColor = activeColor,
```

Comment:
I think making this private property is unnecessary now. Wherever you use `_activeColor`, you can just use `activeColor`. Same with `_trackColor`.

Reply:
I agree, I don't see a benefit for these private variables.

Reply:
Okay, Then can you suggest a way where I can access `activeColor` given by user in constructor  inside getter without using private variable.

Reply:
Similar to the example for text theme: https://github.com/flutter/flutter/blob/6205c110d64351610219d48b52a1a1b9e0140dcd/packages/flutter/lib/src/material/text_theme.dart#L307
You can change `this.activeTrackColor` to `Color? activeTrackColor` then at this line you
```suggestion
        activeTrackColor = activeTrackColor ?? activeColor,
```
Then throughout you only reference `activeTrackColor`. `activeColor` would no longer need to be used.